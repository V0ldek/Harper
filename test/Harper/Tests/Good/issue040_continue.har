nats :: Iterator Integer;
nats = {
  var (i :: Integer) = 0;
  while true {
    yield i;
    i += 1;
  }
};

take :: Integer -> Iterable a -> Iterator a;
take n source = {
  var (i :: Integer) = 0;
  for (x :: a) in source {
    if (i >= n) {
      break;
    }
    yield x;
    i += 1;
  }
};

main :: sideeffect -> ();
main = {
  for (i :: Integer) in take 42 nats {
    if i mod 10 != 0 {
      continue;
    }
    eval printLn i ();
  }
};