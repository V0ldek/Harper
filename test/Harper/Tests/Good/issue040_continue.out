0
10
20
30
40

Execution ended with value: ()